src/main/java/org/genomenexus/vep_wrapper/SystemProcessManager.java

```diff
@@ -0,0 +1,341 @@
+* on which the code is deployed does not support unix style process
+* management through commands 'ps', 'kill', and the java class
+* java.lang.Process
+*/
+public class SystemProcessManager implements Runnable {
+
+    private static Set<Process> launchedProcesses = Collections.synchronizedSet(new HashSet<>());
+    private static String UNIX_PROCESS_CLASS_NAME = "java.lang.UNIXProcess";
+    public static final Long PID_PROBE_WAIT_PERIOD = 2000L; // milliseconds
+    public static final Long REAPER_DAEMON_PERIOD = 250L; // milliseconds
+    public static final int JVM_PROCESS_ID = 1; // maybe this should be tested or determined at runtime
+    public static final int PID_UNAVAILABLE = -1;
+    private static Boolean processIdCanBeReadFromToStringMethod = null; // if available, this avoids reflection
+    private static Boolean processIdCanBeReadFromUnixProcessField = null; // this method depends on an internal field
+    private static Boolean psCommandAvailableOnSystem = null;
+    private static Boolean killCommandAvailableOnSystem = null;
+    private static Field processPidField = null;
+    private static Pattern PID_REPORT_PATTERN = Pattern.compile("pid\\s*=\\s*(\\d+)"); // used to search toString output
+    private static File devNull = null;
+    private static Thread reaperDaemonThread = null;
+    private static final Object reaperDaemonLock = new Object();
+
+    static {
+        devNull = new File("/dev/null"); // used to ignore process data streams
+        try {
+            devNull.exists();
+        } catch (SecurityException e) {
+            devNull = null;
+        }
+    }
+
+    public static Process launchVepProcess(ProcessBuilder pb) {
+        Process p = null;
+        synchronized(reaperDaemonLock) {
+            try {
+                p = pb.start();
+                if (!launchedProcesses.add(p)) {
+                    System.err.println("Warning : attempted to add process to launchedProcesses while it was already present");
+                }
+            } catch (IOException e) {
+                return null;
+            }
+        }
+        startReaperDaemonThreadIfNeeded();
+        return p;
+    }
+
+    public static void destroyVepProcess(Process p) {
+        synchronized(reaperDaemonLock) {
+            Process destroyedProcess = p.destroyForcibly();
+            for (int maximumInterrupts = 1024; maximumInterrupts >= 0; maximumInterrupts--) {
+                try {
+                    destroyedProcess.waitFor();
+                    break;
+                } catch (InterruptedException e) {
+                }
+            }
+            launchedProcesses.remove(p);
+        }
+    }
+
+    public void run() {
+        if (psCommandAvailableOnSystem == null) {
+            determineIfPsCommandIsAvailable();
+        }
+        if (killCommandAvailableOnSystem == null) {
+            determineIfKillCommandIsAvailable();
+        }
+        // ReaperDaemonThread
+        while (true) {
```

Comment:
hesitant to use `true` here as well but at the same time there's a lot going on in this block so maybe just keeping it `true` is best to keep it simple?

Reply:
I have fixed this too. The reaper thread, once started, will continue to run as long as the application is running. So now the condition of the while statement checks whether a shutdown of the reaper was requested and exits if it sees such a request.

I have added a servlet configuration bean which will send the request for reaper shutdown as the servlet is exiting ... and will give a few seconds for the reaper to complete its work before exiting the application.